Multiple issues

jordins opened the pull request.

vHanda committed:
Change default throttle rates to very high values

This used to be done with a sed in the seafile-docker container, but now
we can do this upstream in our fork instead.

vHanda committed:
Move eyeos authentication into seahub

It's easier to have the code over here instead of it being copied from
the docker-seafile image.

vHanda committed:
Add custom modifications

Open365 has modified the theme of seahub and changed its appearance to
meet its requirements. These modifications were originally in the
docker-seafile repo and they were copied during the docker image
creation process.

With this commit the entire code of seahub is now in one place. Instead
of it being distributed across 2 places.

However, this does have the negative affect that this fork of seahub is
strictly for Open365, and no longer an indepdent product.

vHanda committed:
Add our template to TEMPLATE_DIRS

It's so strange that we have this custom seahub-data path which must be
in a particular relative directory.

vHanda committed:
Get customization updates

Our customizations were updated. Please look at the following commits
from docker-seafile to see why -

commit f785133af5582fc5955ff3d8b77bf60bc5b9b8df
commit 26b51b07756110a7f1604e1f60257e13c29499a4
commit e00de2aca1f54ad5a1ab606b15cd6031ec45d046
commit 894962d5020b0c12a0af9f5a37f7f342a6463821
commit c51234105d44fa4a135b85ac4f28d24da885699f
commit ed48e6620da34c72a8f9708b7aa0de6a221aef26

vHanda committed:
Open365 Auth: Load proxy from env variable

Commit be110e232044f71cc3a252ca81775f33f38dd3c3 from docker-seafile

Commit:
Add .idea folder to gitignore

Commit:
Add new extensions for gimp application

Commit:
Remove feature toggling localization download client active

Cotes committed:
Validate cards locally

To avoid the connection between seafileServer and authentication
(through the proxy), we validate card and signature locally with the
same script (/opt/auth/index.js) that we are using for seafdav.

Commit:
Revert "Add new extensions for gimp application"

This reverts commit 48826ba97b9a8e3df77872217d5eeb61febaf862.

vHanda committed:
Add gunicorn access logs

Commit:
Localization text description to access share file public

Commit:
Add localization text description into django.po

jordins committed:
Fix opening files with eyeos

openLinkInEyeosDesktop has a new optional parameter (libraryName).
When calling this function with foreach , the 2nd argument of foreach is
the index inside the array. This results in "libraryName" parameter
always being the index of the array.

This commit fixes it.

Commit:
Extract build command to make development easier.

Commit:
Add height to open365 logo. Limit 32px

Commit:
Add dependencies

Commit:
Add dependencies

Commit:
Associate xcf file extension to be opened with gimp application.

Commit:
Add new icon for edit images with gimp

Commit:
Remove duplicated markup for the edit icon

Commit:
Associate GIMP image files to the default picture file

Commit:
Make the 'edit' button point to the right file

There are some problems when two files with the same name exist.

Commit:
Move getExportedLibraryName function to Common

We'll try to remove code from eyeos.js next.

Commit:
Remove unused function

kampde commented on locale/en/LC_MESSAGES/django.po:
```diff
@@ -7232,3 +7232,23 @@ msgstr "Download client"
+#: seahub/templates/view_shared_dir.html:13
+#: seahub/templates/shared_file_view.html:17
+msgid "File shared using <strong>Open365</strong> personal cloud. "
+msgstr "File shared using <strong>Open365</strong> personal cloud. "
```
`en` translations should leave the `msgstr` blank, as the translation is the same as the key.

kampde commented on locale/en/LC_MESSAGES/django.po:
```diff
@@ -7232,3 +7232,23 @@ msgstr "Download client"
+
+#: seahub/templates/view_shared_dir.html:15
+#: seahub/templates/shared_file_view.html:19
+msgid "Or use it online for free here:  "
```
why lots of keys have extra spacing at the end? It is not usual, I think we should remove them.

kampde commented on media/custom/js/getClipboard.js:
```diff
@@ -0,0 +1,19 @@
+/*
```
remove this useless file.

kampde commented on media/custom/templates/libraries.html:
```diff
@@ -0,0 +1,308 @@
+{% extends "base_for_backbone.html" %}
```
This whole file is duplicated from `seahub/templates/libraries.html` to only change one line ("Welcome to Seafile!" => "Welcome to Open365"). Cannot we extract the name of the product into a variable and use it like it is using `{{ SITE_ROOT }}` in line 21 (and remove this file)?

To check differences: `diff media/custom/templates/libraries.html seahub/templates/libraries.html`

kampde commented on seahub/templates/shared_file_view.html:
```diff
@@ -9,9 +9,17 @@
 
 {% block main_panel %}
+    <div id='publicMessage'>
+        {% trans "File shared using <strong>Open365</strong> personal cloud. " %}
```
please remove trailing spaces from the string to be translated (see comment in the .po files, too).

Cotes committed:
Modifies the generated upload url with a hostname

To understand this commit it is necessary to know the workflow of
seafile to upload a file. First, the client performs a request to the
server with the operation type and a path. When the server receives it,
an URL is generated and sent back to the client. Finally, the client
performs a POST request to the URL with the file attached.

Our problem is that the generated URL in the server-side, uses the
internal private IP. So, the client could not reach the host and, hence,
could not upload the file successfully.

To fix this, when the client performs the first request, we add a new
parameter: hostname. This parameter takes the value of the domain or IP
from the browser URL. Thus, the server will replace the private IP with
the one provided by the client.

Cotes committed:
Merge branch 'open365' of github.com:Open365/seahub into open365

Commit:
Add new function for detect extensions editables on gimp application

Commit:
Revert "Add new function for detect extensions editables on gimp application"

This reverts commit 5349966ba38f1318aafc3041c5e0d2bc446e1c56.

Commit:
Add new file type GIMP

It is for files that one can edit with GIMP.

Commit:
Revert "Revert "Add new function for detect extensions editables on gimp application""

This reverts commit 11de63db3644d8066eb3559bea00e728c1e302cc.

Commit:
Add missing extension to gimp-editable file list

Commit:
Add missing image type to icon list
